pkg/controller/workload/kubernetes/application/application.go

```diff
@@ -281,3 +288,10 @@ func getControllerName(obj metav1.Object) string {
 
 	return c.Name
 }
+
+func condenseErrors(errs []error) error {
+	if len(errs) == 0 {
+		return nil
+	}
+	return fmt.Errorf("%s", errs)
+}
```

Comment:
I think this could be moved to `crossplane-runtime`. I have wanted similar functionality elsewhere before.

Reply:
Aren't there a handful of relatively popular Go libraries that handle this already? I'm not the biggest fan of condensing multiple error together, and tend to try to avoid it as much as possible.